File: HasPairWithSum.go
package main

import (
	"fmt"
)

func main() {
	array1 := []int{1, 5, 4, 49}

	fmt.Println(HasPairWithSum(array1, 5))
}

func HasPairWithSum(arr []int, sum int) bool {
	mySet := make(map[int]bool)
	for _, item := range arr {
		if _, ok := mySet[item]; ok {
			return true
		}
		mySet[sum-item] = true
	}
	return false
}

File: HasPairWithSum.cs
using System;
using System.Collections.Generic;

public class Program
{
    public static void Main()
    {
        string[] array1 = {1, 5, 4, 49}


        Console.WriteLine(HasPairWithSum(array1, 5));
    }

    public static bool HasPairWithSum(int[] arr, int sum) {
        HashSet<int> mySet = new HashSet<int>();
        for (int i = 0; i < arr.Length; i++) {
            if (mySet.Contains(arr[i])) {
                return true;
            }
            mySet.Add(sum - arr[i]);
        }
        return false;
    }
}
